src/main/java/org/neo4j/importer/v1/validation/Neo4jVersionValidator.java

```diff
@@ -0,0 +1,116 @@
+            unsupportedFeatures.add("fulltext_indexes");
+        }
+        if (!isEmpty(schema.getVectorIndexes()) && !neo4jDistribution.hasNodeVectorIndexes()) {
+            unsupportedFeatures.add("vector_indexes");
+        }
+
+        if (!unsupportedFeatures.isEmpty()) {
+            unsupportedPaths.put(String.format("$.targets.nodes[%d].schema", index), unsupportedFeatures);
+        }
+    }
+
+    @Override
+    public void visitRelationshipTarget(int index, RelationshipTarget relationshipTarget) {
+        var schema = relationshipTarget.getSchema();
+        if (schema == null) {
+            return;
+        }
+
+        var unsupportedFeatures = new ArrayList<String>();
+        if (!isEmpty(schema.getTypeConstraints()) && !neo4jDistribution.hasRelationshipTypeConstraints()) {
+            unsupportedFeatures.add("type_constraints");
+        }
+        if (!isEmpty(schema.getKeyConstraints()) && !neo4jDistribution.hasRelationshipKeyConstraints()) {
+            unsupportedFeatures.add("key_constraints");
+        }
+        if (!isEmpty(schema.getUniqueConstraints()) && !neo4jDistribution.hasRelationshipUniqueConstraints()) {
+            unsupportedFeatures.add("unique_constraints");
+        }
+        if (!isEmpty(schema.getExistenceConstraints()) && !neo4jDistribution.hasRelationshipExistenceConstraints()) {
+            unsupportedFeatures.add("existence_constraints");
+        }
+        if (!isEmpty(schema.getRangeIndexes()) && !neo4jDistribution.hasRelationshipRangeIndexes()) {
+            unsupportedFeatures.add("range_indexes");
+        }
+        if (!isEmpty(schema.getTextIndexes()) && !neo4jDistribution.hasRelationshipTextIndexes()) {
+            unsupportedFeatures.add("text_indexes");
+        }
+        if (!isEmpty(schema.getPointIndexes()) && !neo4jDistribution.hasRelationshipPointIndexes()) {
+            unsupportedFeatures.add("text_indexes");
+        }
+        if (!isEmpty(schema.getFullTextIndexes()) && !neo4jDistribution.hasRelationshipFullTextIndexes()) {
+            unsupportedFeatures.add("fulltext_indexes");
+        }
+        if (!isEmpty(schema.getVectorIndexes()) && !neo4jDistribution.hasRelationshipVectorIndexes()) {
+            unsupportedFeatures.add("vector_indexes");
+        }
+
+        if (!unsupportedFeatures.isEmpty()) {
+            unsupportedPaths.put(String.format("$.targets.relationships[%d].schema", index), unsupportedFeatures);
+        }
+    }
+
+    @Override
+    public boolean report(SpecificationValidationResult.Builder builder) {
+        unsupportedPaths.forEach((path, features) -> builder.addError(
+                path,
+                ERROR_CODE,
+                String.format("%s, features are not supported by %s.", features, neo4jDistribution.toString())));
+        return !unsupportedPaths.isEmpty();
+    }
+
+    private boolean isEmpty(List<?> list) {
+        return list == null || list.isEmpty();
```

Comment:
```suggestion
        return list.isEmpty();
```

let's make sure none of the lists are `null` (if needed, you can change the getters to return empty lists)

Reply:
👍